src/Transfer/Destinations/Local.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Utopia\Transfer\Destinations;
+
+use Utopia\Transfer\Destination;
+use Utopia\Transfer\Resources\Storage\File;
+use Utopia\Transfer\Resources\Storage\FileData;
+use Utopia\Transfer\Resources\Functions\Deployment;
+use Utopia\Transfer\Resource;
+use Utopia\Transfer\Transfer;
+
+/**
+ * Local
+ *
+ * Used to export data to a local file system or for testing purposes.
+ * Exports all data to a single JSON File.
+ */
+class Local extends Destination
+{
+    private array $data = [];
+
+    protected string $path;
+
+    public function __construct(string $path)
+    {
+        $this->path = $path;
+
+        if (!\file_exists($this->path)) {
+            mkdir($this->path, 0777, true);
+            mkdir($this->path . '/files', 0777, true);
+            mkdir($this->path . '/deployments', 0777, true);
+        }
+    }
+
+    /**
+     * Get Name
+     *
+     * @return string
+     */
+    public static function getName(): string
+    {
+        return 'Local';
+    }
+
+    /**
+     * Get Supported Resources
+     *
+     * @return array
+     */
+    public function getSupportedResources(): array
+    {
+        return [
+            Resource::TYPE_ATTRIBUTE,
+            Resource::TYPE_BUCKET,
+            Resource::TYPE_COLLECTION,
+            Resource::TYPE_DATABASE,
+            Resource::TYPE_DOCUMENT,
+            Resource::TYPE_FILE,
+            Resource::TYPE_FILEDATA,
+            Resource::TYPE_FUNCTION,
+            Resource::TYPE_DEPLOYMENT,
+            Resource::TYPE_HASH,
+            Resource::TYPE_INDEX,
+            Resource::TYPE_USER,
+            Resource::TYPE_ENVVAR,
+            Resource::TYPE_TEAM,
+            Resource::TYPE_TEAM_MEMBERSHIP,
+        ];
+    }
+
+    public function report(array $resources = []): array
+    {
+        $report = [];
+
+        if (empty($resources)) {
+            $resources = $this->getSupportedResources();
+        }
+
+        // Check we can write to the file
+        if (!\is_writable($this->path . '/backup.json')) {
+            $report[Transfer::GROUP_DATABASES][] = 'Unable to write to file: ' . $this->path;
+            throw new \Exception('Unable to write to file: ' . $this->path);
+        }
+
+        return $report;
+    }
+
+    public function syncFile(): void
```

Comment:
Why are we exposing this method publicly? Can we just name it `sync`? In this context, it should be clear enough.